### DISCUSSION TITLE
RuntimeError: Camera __init__ sequence did not complete.

### DISCUSSION THREAD

**Discussion by 8JupiterMoll8** [raspberrypi/picamera2#702] [General]
Hello,

I'm currently engaged in audio-video recording using Flask in a browser-based environment. I have encountered an issue where, if the camera is not properly stopped or closed during AV recording, I receive a RuntimeError stating, "Camera init sequence did not complete." This typically happens when I switch back to another route in Flask.

I'm wondering if there's a way to catch this error using an exception handler, allowing me to execute the camera.close() function. This would allow the camera to be appropriately closed and restarted. And the Flask app would not crashed. 

Is this the right way to go?

```
try:
    # Code that may raise the RuntimeError
    # For example, camera initialization code
    # ...
except RuntimeError as e:
    if str(e) == "Camera __init__ sequence did not complete.":
        # Handle the specific error
        camera.close()
        print("Camera i.")

```

**Comment by davidplowman** [comment]
Hi, I think that may work OK. Maybe you could also try using a `with` statement, which should clean up automatically for you:
```
with Picamera2() as camera:
    # do stuff...
```
